### PR TITLE
Shadow service typename filter

### PR DESCRIPTION
**Pull request by gnawf**

### CODE REVIEW

**Comment by Juliano-Prado** [comment]
Code look good to me but I missed some test showing side effect of when isShadowingEnabled.

**Review by Juliano-Prado — APPROVED** [review]
Code look good to me but I missed some test showing side effect of when isShadowingEnabled.

File: lib/src/main/java/graphql/nadel/hints/NadelServiceTypenameShadowingMismatchHint.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+            }
+
+            override fun onMismatch(
+                oldDecision: Boolean,
+                newDecision: Boolean,
+                service: Service,
+                objectTypeName: String,
+                field: ExecutableNormalizedField,
+            ) {
+            }
```
Comment (Juliano-Prado) [inline lib/src/main/java/graphql/nadel/hints/NadelServiceTypenameShadowingMismatchHint.kt:30]: is it right to be empty}

**Review by Juliano-Prado — COMMENTED** [review]
Code look good to me but I missed some test showing side effect of when isShadowingEnabled.

File: lib/src/main/java/graphql/nadel/hints/NadelServiceTypenameShadowingMismatchHint.kt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        field: ExecutableNormalizedField,
+    )
+
+    companion object {
+        internal val default = object : NadelServiceTypenameShadowingHint {
+            override fun isShadowingEnabled(): Boolean {
+                return false
+            }
+
+            override fun onMismatch(
```
Comment (Juliano-Prado) [inline lib/src/main/java/graphql/nadel/hints/NadelServiceTypenameShadowingMismatchHint.kt:23]: default implementation is empty, I got lost on how is setting behaviour when it is called on [NadelServiceTypeFilterTransform.kt](https://github.com/atlassian-labs/nadel/pull/637/files#diff-ef65360fa390e6d9ac0bf031a350bd2c8138040cf3664acfc1a9d98da8f1107d)

Comment (gnawf) [reply lib/src/main/java/graphql/nadel/hints/NadelServiceTypenameShadowingMismatchHint.kt:23]: The hint is provided by AGG. So the impl will be logging the mismatch.

**Pull request merged**